### PR TITLE
Vsfp 91 abstract

### PR DESCRIPTION
**Pull request by lstreckeisen**

### CODE REVIEW

File: SA/design_concept/abstract.typ [hunk 1]
```diff
@@ -1,5 +1,22 @@
 = Abstract
-Content: Problem description, goal, proceedure, results, recommendations
-(Keywords?)
+Most visual programming tools used to introduce children & young adults to the world of programming are made for imperative programming.
+Existing tools for functional programming lack in visual aesthetics or hide key concepts of the functional programming paradigm.
+
+To address this gap, a visual, block-based tool for functional programming should be designed.
+The focus of the project lies on finding a visual design of such a tool and then prove its feasibility in a proof of concept application.
```
Comment (fl3pp) [inline SA/design_concept/abstract.typ:6]:
```suggestion
The project focuses on finding a visual design of such a tool and then proving its feasibility in a proof of concept application.
```

File: SA/design_concept/abstract.typ [hunk 2]
```diff
@@ -1,5 +1,22 @@
...
+
+Before creating a visual design, existing visual programming tools are examined.
```
Comment (fl3pp) [inline SA/design_concept/abstract.typ:8]:
```suggestion
Existing visual programming tools are examined before creating a visual design.
```

File: SA/design_concept/abstract.typ [hunk 3]
```diff
@@ -1,5 +1,22 @@
...
+The design development consists of two iterations:
+The first concept drafts are based on researched tools and evaluated using a survey.
+Using the received feedback, a new concept is created and implemented in a PoC.
```
Comment (fl3pp) [inline SA/design_concept/abstract.typ:11]:
```suggestion
The development of the design is then approached in two iterations: 
In the first step, concept drafts are based on researched tools and evaluated using a survey.
Then, a new concept is created and implemented in a PoC using the received feedback.
```

Comment (fl3pp) [reply SA/design_concept/abstract.typ:11]: Is 'PoC' ok to use here in the abstract?

It's probably the first use of the abbreviation in text, but the abstract also is a bit special, so I don't know.

File: SA/design_concept/abstract.typ [hunk 4]
```diff
@@ -1,5 +1,22 @@
...
+
+The created design concept is focused on function composition, guided by type holes that indicate the type required to receive a valid function definition.
```
Comment (fl3pp) [inline SA/design_concept/abstract.typ:13]:
```suggestion
The created design concept focuses on function composition, guided by type holes that indicate the type required for a valid function definition.
```

File: SA/design_concept/abstract.typ [hunk 5]
```diff
@@ -1,5 +1,22 @@
...
+The PoC includes an inference engine that determines the type of undefined parts of a function and is built using Electron.js & Haskell.
+
+The PoC proves that the proposed concept for function composition works as envisioned.
+To create a visual tool that can be used in a classroom, it is recommended to conduct an additional project to find a visual design for data type construction.
+In addition, some changes in the used technologies are recommended.
```
Comment (fl3pp) [inline SA/design_concept/abstract.typ:18]:
```suggestion

The PoC proves that the proposed concept for function composition works as envisioned. 
It includes an inference engine that determines the type of undefined parts of a function and is built using Electron.js & Haskell.

It is recommended to conduct an additional project to find a visual design for data type construction to create a visual tool that can be used in a classroom.
In addition, some changes in the used technologies are recommended.
```

File: SA/design_concept/abstract.typ (hunk 3 above)
Comment (lstreckeisen) [reply SA/design_concept/abstract.typ]: I've no idea. I replaced it.

File: SA/design_concept/abstract.typ [hunk 6]
```diff
@@ -1,5 +1,23 @@
...
-Content: Problem description, goal, proceedure, results, recommendations
-(Keywords?)
+Most visual programming tools used to introduce children & young adults to the world of programming are made for imperative programming.
+Existing tools for functional programming lack in visual aesthetics or hide key concepts of the functional programming paradigm.
+
+To address this gap, a visual, block-based tool for functional programming should be designed.
+The project focuses on finding a visual design of such a tool and then proving its feasibility in a proof of concept application.
+
+Existing visual programming tools are examined before creating a visual design.
+The development of the design is approachedin two iterations:
```
Comment (fl3pp) [inline SA/design_concept/abstract.typ:9]:
```suggestion
The development of the design is approached in two iterations:
```

**Review by fl3pp — APPROVED** [review]

**Pull request merged**